### PR TITLE
[docs] Add a sequence diagram with api calls, and recommended app patterns

### PR DESCRIPTION
**Pull request by geoknee**

### CODE REVIEW

**Comment by geoknee** [comment]
See https://5f195e48a7719101a58a7dea--nitro-protocol.netlify.app/docs/next/app-devs/make-api-calls and https://5f195e48a7719101a58a7dea--nitro-protocol.netlify.app/docs/next/app-devs/recommended-patterns

File: packages/nitro-protocol/docs/app-devs/make-api-calls.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+        ClientB->>WalletB: p.enable();
+        ClientB->>WalletB: onMessageQueued(callback)
+        ClientB->>WalletB: onChannelProposed(callback)
+        ClientB->>WalletB: onChannelUpdated(callback)
+    end
+rect rgba(0, 0, 255, .1)
+    note left of WalletB: Opening a channel
+    ClientA->>WalletA: createChannel();
+    WalletA-->>ClientA: ChannelUpdated('proposed')
+    WalletA-->>ClientA: MessageQueued(msg)
```
Comment (tomclose) [inline packages/nitro-protocol/docs/app-devs/make-api-calls.md:37]: `msg0`?

File: packages/nitro-protocol/docs/app-devs/make-api-calls.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    ClientA->>ClientB: msg0
+    ClientB->>WalletB: pushMessage(msg0)
+    WalletB-->>ClientB: ChannelProposed('id')
+    ClientB->>+WalletB: joinChannel('id');
+    WalletB-->>ClientB: ChannelUpdated('funding');
+    WalletB-->>ClientB: MessageQueued(msg)
```
Comment (tomclose) [inline packages/nitro-protocol/docs/app-devs/make-api-calls.md:43]: `msg1`?

File: packages/nitro-protocol/docs/app-devs/recommended-patterns.md [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+
+For a simple game like Rock Paper Scissors, the advantages of virtual funding may or may not be considered necessary. Regardless, switching from direct-funding to virtual-funding at a later stage should be straightforward: statechannel execution should be completely independent of the funding mechanism.
+
+## Application coding
+
+In order for your app to construct state updates of the right format, you'll need to write some javascript / typescript helper code. The purpose of this code is to translate between the general-purpose, [protocol-level concepts of `appData` and `outcome`](./quick-start-contracts) and the specific data structures that your application needs to work with. Those structures should mirror those in the olidity code you have already written. Here's an idea of what we did in Rock Paper Scissors (see the solidity code [here](https://github.com/statechannels/apps/blob/master/packages/rps/contracts/RockPaperScissors.sol)):
```
Comment (tomclose) [inline packages/nitro-protocol/docs/app-devs/recommended-patterns.md:30]: `olidity` -> `solidity`

File: packages/nitro-protocol/docs/app-devs/recommended-patterns.md [hunk 4]
```diff
@@ -0,0 +1,284 @@
...
+```
+
+## Wrapping the channelClient
+
+We recommend that you wrap the channelClient in an `AppSecificChannelClient`, which can leverage your app-specific coding [as above](#application-coding). This additional layer of abstraction should result in cleaner code for the rest of your application.
+
+Here's a sketch of what we did with Rock Paper Scissors:
+
+```typescript
+export class RPSChannelClient {
```
Comment (tomclose) [inline packages/nitro-protocol/docs/app-devs/recommended-patterns.md:192]: This doesn't look like it has all the functionality?

Comment (geoknee) [reply packages/nitro-protocol/docs/app-devs/recommended-patterns.md:192]: No, I stripped most of it out. There's already a lot of code on this page, and I didn't want to overdo it. I also wanted to reflect the fact that we don't (yet) have a very rigid or fixed set of patterns for these wrapping classes. Hence "sketch".

Comment (geoknee) [reply packages/nitro-protocol/docs/app-devs/recommended-patterns.md:192]: Have added a placeholder to make it clearer that the example is incomplete, as well as a GH link to the full thing.

**Pull request merged**